SEBrowser/Controllers/OpenXDA/TrendChannelController.cs

```diff
@@ -0,0 +1,353 @@
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  04/05/2023 - Gabriel Santos
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+using GSF.Data;
+using GSF.Data.Model;
+using GSF.Identity;
+using GSF.Web.Model;
+using openXDA.Model;
+using SystemCenter.Model;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Net;
+using System.Web.Http;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using System.Net.Http;
+using openXDA.APIAuthentication;
+using Newtonsoft.Json;
+using System.Text;
+using System.IO;
+
+namespace SEBrowser.Controllers.OpenXDA
+{
+    [RoutePrefix("api/OpenXDA")]
+    public class TrendChannelController : ApiController
+    {
+        #region [ Nested Types ]
+
+        // ToDo: These next two are defined in the HIDS controller for xda, do we want to make that an import?
+        private class HistogramMetadata
+        {
+            public int ChannelID { get; set; }
+            public int FundamentalFrequency { get; set; }
+            public int SamplingRate { get; set; }
+            public DateTime StartTime { get; set; }
+            public DateTime EndTime { get; set; }
+            public int TotalCapturedCycles { get; set; }
+            public double CyclesMax { get; set; }
+            public double CyclesMin { get; set; }
+            public double ResidualMax { get; set; }
+            public double ResidualMin { get; set; }
+            public double FrequencyMax { get; set; }
+            public double FrequencyMin { get; set; }
+            public double RMSMax { get; set; }
+            public double RMSMin { get; set; }
+            public int CyclicHistogramBins { get; set; }
+            public int ResidualHistogramBins { get; set; }
+            public int FrequencyHistogramBins { get; set; }
+            public int RMSHistogramBins { get; set; }
+        }
+        private class HistogramPoint
+        {
+            public int Bin { get; set; }
+
+            public int Sample { get; set; }
+
+            public float Value { get; set; }
+        }
+        private class LinearPoint
```

Comment:
can we use `Hids.Point` instead? since these are actually coming from the openXDA which uses `Hids.Point` when creating the stream?

Reply:
We could, but I don't believe HIDS.Point from XDA is currently exported.

Reply:
As of a recent update, this class has been removed.